src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use DateTimeImmutable;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class PromotionRepository extends EntityRepository
+{
+    /**
+     * @return array[]
+     */
+    public function findActivePromotionsByPid(Pid $pid, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('promotion')
+            ->addSelect(['context', 'promotionOfCoreEntity', 'promotionOfImage'])
```

Comment:
Will we be using the Context when building the promotion object? If not you can exclude it from the select clause